torch/csrc/jit/custom_operator.h

```diff
@@ -204,21 +199,54 @@ Operator createOperator(
       c10::guts::typelist::map_t<decay_t, typename Traits::parameter_types>;
   using ArgumentTuple =
       typename c10::guts::typelist::to_tuple<ArgumentTypes>::type;
-  using ReturnType = decay_t<typename Traits::return_type>;
 
   auto schema = torch::jit::detail::inferAndCheckSchema<Traits>(schemaOrName);
 
   return Operator(schema, [implementation, schema](Stack& stack) {
     ArgumentTuple tuple;
-    auto result = torch::jit::detail::callOperatorWithTuple<ReturnType>(
+    torch::jit::detail::callOperatorWithTuple(
         schema,
         std::move(implementation),
         stack,
         tuple,
         typename MakeIndices<std::tuple_size<ArgumentTuple>::value>::indices{});
-    pack(stack, std::move(result));
     return 0;
   });
 }
+
+/// Registration class for new operators. Effectively calls
+/// `torch::jit::registerOperator` for every supplied operator, but allows doing
+/// so in the global scope when a `RegisterOperators` object is assigned to a
+/// static variable. Also handles registration of user-defined, "custom"
+/// operators.
+struct TORCH_API RegisterOperators {
```

Comment:
it's a bit weird that it's plural but used usually to register a single op

Reply:
The original code using this, e.g. `register_aten_ops.cpp`, looks more like this:
```
RegisterOperators reg({
  Operator(
      "aten::RoiPooling2d_backward(Tensor input, Tensor rois, int pooledHeight, int pooledWidth, float spatialScale, Tensor gradOutput, Tensor argmaxes) -> Tensor",
      [](Stack & stack) {
          autograd::profiler::RecordFunction record("RoiPooling2d_backward");
          auto result = at::RoiPooling2d_backward(
              (std::move(peek(stack, 0, 7))).toTensor(),
              (std::move(peek(stack, 1, 7))).toTensor(),
              (std::move(peek(stack, 2, 7))).toInt(),
              (std::move(peek(stack, 3, 7))).toInt(),
              (std::move(peek(stack, 4, 7))).toDouble(),
              (std::move(peek(stack, 5, 7))).toTensor(),
              (std::move(peek(stack, 6, 7))).toTensor()
          );
          drop(stack, 7);
          pack(stack, std::move(result));
          return 0;
      }
  ),
  Operator(
      "aten::RoiPooling2d_forward(Tensor input, Tensor rois, int pooledHeight, int pooledWidth, float spatialScale) -> (Tensor, Tensor)",
      [](Stack & stack) {
          autograd::profiler::RecordFunction record("RoiPooling2d_forward");
          auto result = at::RoiPooling2d_forward(
              (std::move(peek(stack, 0, 5))).toTensor(),
              (std::move(peek(stack, 1, 5))).toTensor(),
              (std::move(peek(stack, 2, 5))).toInt(),
              (std::move(peek(stack, 3, 5))).toInt(),
              (std::move(peek(stack, 4, 5))).toDouble()
          );
          drop(stack, 5);
          pack(stack, std::move(result));
          return 0;
      }
  ),
  Operator(
      "aten::__and__(Tensor self, Tensor other) -> Tensor",
      [](Stack & stack) {
          autograd::profiler::RecordFunction record("__and__");
          auto result = at::__and__(
              (std::move(peek(stack, 0, 2))).toTensor(),
              (std::move(peek(stack, 1, 2))).toTensor()
          );
          drop(stack, 2);
          pack(stack, std::move(result));
          return 0;
      }
  ),
...
```